### PR TITLE
feat: ability to require an email

### LINKED ISSUE #18047: [CAL-4857] ability to require an email
right now we have “exclude”.

lets the opposite for people who wanna lock down their form and require a certain email

Error:

* Your email has no permission to book. contact the owner of this link.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4857](https://linear.app/calcom/issue/CAL-4857/ability-to-require-an-email)</sub>

### PR DESCRIPTION
**Pull request by kart1ka**
## What does this PR do?

- Fixes #18047 (GitHub issue number)
- Fixes CAL-4857 (Linear issue number - should be visible at the bottom of the GitHub issue description)

https://github.com/user-attachments/assets/7c99fa08-c2b4-481e-9c8f-5b306a065533

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: packages/features/form-builder/FormBuilder.tsx [hunk 1]
```diff
@@ -574,13 +574,34 @@ function FieldEditDialog({
...
+                          } catch (err) {
+                            if (err instanceof ZodError) {
+                              fieldForm.setError("requireEmails", {
+                                message: err.errors[0]?.message || "Invalid input",
+                              });
+                            }
+                          }
+                        }}
+                        label={t("require_emails_that_contain")}
+                        placeholder="gmail.com, hotmail.com, ..."
```
Comment (Praashh) [inline packages/features/form-builder/FormBuilder.tsx:594]:
```suggestion
                        placeholder="gmail.com, hotmail.com, exclude@gmail.com, ..."
```

**Review by Praashh — DISMISSED** [review]
LGTM!!

File: apps/web/public/static/locales/en/common.json [hunk 2]
```diff
@@ -2825,7 +2825,9 @@
   "rr_distribution_method_balanced_title": "Load balancing",
   "rr_distribution_method_balanced_description": "We will monitor how many bookings have been made with each host and compare this with others, disabling some hosts that are too far ahead so bookings are evenly distributed.",
   "exclude_emails_that_contain": "Exclude emails that contain ...",
+  "require_emails_that_contain": "Require emails that contain ...",
   "exclude_emails_match_found_error_message": "Please enter a valid work email address",
+  "require_emails_no_match_found_error_message": "Your email has no permission to book. Contact the owner of this link.",
```
Comment (keithwillcode) [inline apps/web/public/static/locales/en/common.json:2830]:
```suggestion
  "require_emails_no_match_found_error_message": "You are not allowed to book with this email domain. Contact the owner of this link.",
```

File: apps/web/public/static/locales/en/common.json [hunk 3]
```diff
@@ -2825,7 +2825,9 @@
   "rr_distribution_method_balanced_title": "Load balancing",
   "rr_distribution_method_balanced_description": "We will monitor how many bookings have been made with each host and compare this with others, disabling some hosts that are too far ahead so bookings are evenly distributed.",
   "exclude_emails_that_contain": "Exclude emails that contain ...",
+  "require_emails_that_contain": "Require emails that contain ...",
   "exclude_emails_match_found_error_message": "Please enter a valid work email address",
+  "require_emails_no_match_found_error_message": "You are not allowed to book with this email domain. Contact the owner of this link.",
```
Comment (keithwillcode) [inline apps/web/public/static/locales/en/common.json:2830]:
```suggestion
  "require_emails_no_match_found_error_message": "You are not allowed to book with this email domain. Please contact the owner of this link.",
```

**Review by keithwillcode — DISMISSED** [review]
Tests should be added to `packages/features/bookings/lib/getBookingResponsesSchema.test.ts`

**Comment by kart1ka** [comment]
> Tests should be added to `packages/features/bookings/lib/getBookingResponsesSchema.test.ts`

Added tests.

**Review by anikdhabal — APPROVED** [review]
LGTM!!

**Pull request merged**